catkin_ws/src/vision/src/object_detection.py

```diff
@@ -71,7 +71,7 @@ def detect_on_image(raw_img, camera_id):
             if camera_id == 0: # DOWN CAM
                 if global_class_name == "Lane Marker":
                     detectionFrame.label = global_class_name
-                    confidence.append(conf)
+                    detectionFrame.confidence = conf
```

Comment:
I think you forgot to delete the initialization of the confidence array (line 46)